lib/db/structs.go

```diff
@@ -365,6 +370,32 @@ func (vl VersionList) findDevice(device []byte) (bool, int, int, bool) {
 	return false, -1, -1, false
 }
 
+func (vl VersionList) popVersion(version protocol.Vector) (VersionList, FileVersion, bool) {
+	i := vl.versionIndex(version)
+	if i == -1 {
+		return vl, FileVersion{}, false
+	}
+	fv := vl.RawVersions[i]
+	return vl.popVersionAt(i), fv, true
+}
+
+func (vl VersionList) versionIndex(version protocol.Vector) int {
+	for i, v := range vl.RawVersions {
+		if version.Equal(v.Version) {
+			return i
+		}
+	}
+	return -1
+}
+
+func (vl VersionList) popVersionAt(i int) VersionList {
```

Comment:
This (and maybe the others) feel like they should be pointer methods, as it modifies the receiver. I didn't check if that's against expectations anywhere (I think not) but it otherwise looks like a value method which it isn't because the receiver is actually invalid after the pop.

Reply:
I honestly just blindly followed how it previously worked, and it was always in the style of `(vl VersionList) update(...) VersionList`, i.e. value receiver and return the value. Changing to pointer receiver seems sensible, but then there's no point in returning the receiver - or am I missing something here?